### ISSUE TITLE
Invalid CA1508 dead conditional code diagnostic on using statement

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn-analyzers#4728]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3

### Describe the bug

Seen here in the .NET Project System:

https://github.com/dotnet/project-system/blob/8e23c27f0e251f47094caf6822b435a7b44b125c/src/Microsoft.VisualStudio.AppDesigner/ApplicationDesigner/ApplicationDesignerEditorFactory.vb#L83

```vb
Using New Common.WaitCursor
```

This type sets the wait cursor, and undoes the change on disposal.

### Expected behavior

No diagnostic. There is no conditional here. There is no dead code here.

### Actual behavior

Diagnostic.

![image](https://user-images.githubusercontent.com/350947/105444291-a87add80-5cc1-11eb-91fc-d5d899ed84cd.png)

**Comment by drewnoakes** [comment]
Another example in VB.NET code, also related to a `Using` statement:

![image](https://user-images.githubusercontent.com/350947/105444685-6aca8480-5cc2-11eb-9393-46d5b460f13d.png)

**Comment by Youssef1313** [comment]
A similar issue: https://github.com/dotnet/roslyn-analyzers/issues/4526 was claimed to be fixed in 5.0.1.

**Comment by Youssef1313** [comment]
I attempted to use 5.0.3 in https://github.com/dotnet/project-system/pull/6855, but this false positive is still there unfortunately.

**Comment by Youssef1313** [comment]
Ah, found it.

https://github.com/dotnet/project-system/blob/39a73df453543ed68120066a8f156775f06a387b/build/import/Packages.targets#L113

It's still using the deprecated package.

**Comment by mavasani** [comment]
Thanks @Youssef1313.

@drewnoakes You can follow steps at https://docs.microsoft.com/visualstudio/code-quality/migrate-from-fxcop-analyzers-to-net-analyzers to migrate to newer analyzer package.

**Comment by drewnoakes** [comment]
Thanks. I found those docs easily enough (as feedback on the usability of the change between packages). For this CA there were loads of false positives beyond the one mentioned here (such was failures in definite assignment analysis), so I disabled it.

Locally I have an attempt at not just upgrading the package but also setting AllEnabledByDefault and latest analysis level, which produced ~600 diagnostics, which I'm working through.

**Comment by Youssef1313** [comment]
> For this CA there were loads of false positives beyond the one mentioned here (such was failures in definite assignment analysis)

I see most of them are around `Debug.Assert`s followed by an `if` condition. Will try to write a minimal repro soon.

**Comment by mavasani** [comment]
I am closing this out as the original reported issue with using statement has been fixed. Please file separate issue(s) for other false positives.
